### PR TITLE
Add Android network interface change documentation.

### PR DESCRIPTION
**Pull request by jonesc-oci**

### CODE REVIEW

**Comment by iguessthislldo** [comment]
So users need to create JNI calls for these themselves?

File: docs/android.md [hunk 1]
```diff
@@ -446,7 +446,30 @@ using the
 but the exact method for doing so will depend on the API level and the needs of
 the app.
```
Comment (iguessthislldo) [inline docs/android.md]: This paragraph doesn't really apply any more, can you remove it?

File: docs/android.md [hunk 2]
```diff
@@ -446,7 +446,30 @@ using the
...
+                super.onAvailable(network);
+                // ...
+            }
+
+            @Override
+            public void onLost(Network network) {
+                super.onLost(network);
+                // ...
+            }
+        });
```
Comment (iguessthislldo) [inline docs/android.md]:
```suggestion
    cm.registerDefaultNetworkCallback(new ConnectivityManager.NetworkCallback() {
        @Override
        public void onAvailable(Network network) {
            super.onAvailable(network);
            // ...
        }

        @Override
        public void onLost(Network network) {
            super.onLost(network);
            // ...
        }
    });
```

**Comment by iguessthislldo** [comment]
Also are you documenting Android Service usage in this PR?

**Comment by mitza-oci** [comment]
Azure CI builds cancelled since they're not relevant for Markdown changes.

**Comment by jonesc-oci** [comment]
> So users need to create JNI calls for these themselves?

No, developer's shouldn't have to write JNI for the Network change notifications. 

> Also are you documenting Android Service usage in this PR?

I was only documenting the NetworkConfigMonitor, but I will also update the Android Service comments around line 597.

**Comment by iguessthislldo** [comment]
> No, developer's shouldn't have to write JNI for the Network change notifications.

Okay, if that's not it, what is the user supposed to do inside the `onAvailable` and `onLost` callbacks? I'm imagining there is some way the user would have to interact with OpenDDS to tell it that the networks have changed.

**Comment by jonesc-oci** [comment]
OpenDDS network connections change automatically since Android supports netlink sockets. The example shows how an app can receive callback notifications of a network change. It's unrelated to OpenDDS, so I will remove it.

**Pull request merged**